starknet_py/abi/v1/parser_transformer.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+import lark
+from lark import Transformer, v_args
+
+from starknet_py.cairo.data_types import ArrayType, CairoType, FeltType, Option
+
+
+class ParserTransformer(Transformer):
+    """
+    Transforms the lark tree into a CairoTypes.
+    """
+
+    # pylint: disable=no-self-use
+
+    def __default__(self, data: str, children, meta):
+        raise TypeError(f"Unable to parse tree node of type {data}")
+
+    @v_args(inline=True)
+    def start(self, value):
+        return value
+
+    def type(self, value):
+        return value[0]
```

Comment:
Why index 0?

Reply:
Because value is a list of types

Reply:
and it's length is 1 here

Reply:
Maybe let's explicitly check value is of length 1? To make it extra clear.

Reply:
I don't think `abi.ebnf` lets the value be longer than 1 here, but I can check it for extra clarity